quinn-proto/src/connection/streams.rs

```diff
@@ -1034,6 +1040,39 @@ impl Recv {
         }
     }
 
+    /// Returns the window that should be advertised in a `MAX_STREAM_DATA` frame
+    ///
+    /// The method returns a tuple which consists of the window that should be
+    /// announced, as well as a boolean parameter which indicates if a new
+    /// transmission of the value is recommended. If the boolean value is
+    /// `false` the new window should only be transmitted if a previous transmission
+    /// had failed.
+    fn max_stream_data(&mut self, stream_receive_window: u64) -> (u64, bool) {
+        let max_stream_data = self.assembler.bytes_read() + stream_receive_window;
+
+        // Only announce a window update if it's significant enough
+        // to make it worthwhile sending a MAX_STREAM_DATA frame.
+        // We use here a fraction of the configured stream receive window to make
+        // the decision, and accomodate for streams using bigger windows requring
+        // less updates. A fixed size would also work - but it would need to be
+        // smaller than `stream_receive_window` in order to make sure the stream
+        // does not get stuck.
+        let diff = max_stream_data - self.transmitted_max_stream_data;
+        let transmit = self.receiving_unknown_size() && diff >= (stream_receive_window / 8);
+        (max_stream_data, transmit)
+    }
+
+    /// Records that a `MAX_STREAM_DATA` announcing a certain window was transmitted
+    ///
+    /// This will suppress enqueuing further `MAX_STREAM_DATA` frames unless
+    /// either the previous transmission was not acknowledged or the window
+    /// further increased.
+    pub fn record_transmitted_max_stream_data(&mut self, transmitted_value: u64) {
```

Comment:
I find the `record_` prefix here pretty superfluous. Otherwise this seems fine.

Reply:
Well it should have some prefix, since it’s not a getter. And I don’t prefer using `set_` for this. Since that one sounds like you can always call it with an arbitrary value.

Reply:
Maybe replace `transmitted` with `sent`? I just find `record_transmitted_max_stream_data` very wordy.

Reply:
I updated it to `sent`. 

However I personally don't think anything wordy or verbose in code is bad, as long as it helps readers to understand what is meant.